src/arrayMethodJoin.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedItems = '';
+    let separatorChar = separator;
+
+    if (typeof separatorChar === 'undefined') {
+      separatorChar = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined
+        || this[i] === null) {
+        joinedItems += '';
+      } else {
+        joinedItems += this[i];
+      }
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== undefined
        || this[i] !== null) {
        joinedItems += this[i];
      }
```
вот так вот можно немного сократить код)

Reply:
только тогда там вместо || &&